x-pack/plugins/apm/public/components/app/service_logs/index.tsx

```diff
@@ -0,0 +1,102 @@
+ * 2.0.
+ */
+
+import { EuiLoadingSpinner, EuiEmptyPrompt } from '@elastic/eui';
+import React, { useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import moment from 'moment';
+import { useUrlParams } from '../../../context/url_params_context/use_url_params';
+import { FETCH_STATUS, useFetcher } from '../../../hooks/use_fetcher';
+import { useApmServiceContext } from '../../../context/apm_service/use_apm_service_context';
+import { LogStream } from '../../../../../infra/public';
+
+export function ServiceLogs() {
+  const { serviceName } = useApmServiceContext();
+  const {
+    urlParams: { environment, kuery, start, end },
+  } = useUrlParams();
+
+  const { data, status } = useFetcher(
+    (callApmApi) => {
+      if (start && end) {
+        return callApmApi({
+          endpoint: 'GET /api/apm/services/{serviceName}/infrastructure',
+          params: {
+            path: { serviceName },
+            query: {
+              environment,
+              kuery,
+              start,
+              end,
+            },
+          },
+        });
+      }
+    },
+    [environment, kuery, serviceName, start, end]
+  );
+
+  const noInfrastructureData = useMemo(() => {
+  return (
+    isEmpty(data?.serviceInfrastructure?.containerIds) &&
+    isEmpty(data?.serviceInfrastructure?.hostNames) &&
+    isEmpty(data?.serviceInfrastructure?.podNames)
+  );
+  }, [data]);
+
+  if (status === FETCH_STATUS.LOADING) {
+    return (
+      <div style={{ textAlign: 'center' }}>
+        <EuiLoadingSpinner size="m" />
+      </div>
+    );
+  }
+
+  if (status === FETCH_STATUS.SUCCESS && noInfrastructureData) {
+    return (
+      <EuiEmptyPrompt
+        title={
+          <h2>
+            {i18n.translate('xpack.apm.serviceLogs.noInfrastructureMessage', {
+              defaultMessage: 'There are no log messages to display.',
+            })}
+          </h2>
+        }
+      />
+    );
+  }
+  
+  return (
+      <LogStream
+      columns={[{ type: 'timestamp' }, { type: 'message' }]}
+      height={'60vh'}
+      startTimestamp={moment(start).valueOf()}
+      endTimestamp={moment(end).valueOf()}
+      query={`
+       ${
+         (!!data?.serviceInfrastructure?.containerIds?.length)
+           ? data.serviceInfrastructure.containerIds
+               .map((id) => `${CONTAINER_ID}: "${id}"`)
+               .join(' or ')
+           : ''
+       } ${
+        (data?.serviceInfrastructure?.hostNames?.length ?? 0) > 0
+          ? `or host.name : ${data?.serviceInfrastructure.hostNames
+              .map((hostName) => `"${hostName}"`)
+              .join(' or host.name : ')}`
+          : ''
+      } ${
+        (data?.serviceInfrastructure?.podNames?.length ?? 0) > 0
+          ? `or kubernetes.pod.name : ${data?.serviceInfrastructure.podNames
+              .map((podName) => `"${podName}"`)
+              .join(' or kubernetes.pod.name : ')}`
+          : ''
+      }
```

Comment:
I should have been more clear. Since you updated the first, please also update the two others here for consistency

Reply:
Yeah, I'm doing it locally now. I've actually moved this into a separate function as even with this syntax you still need an `or` to (potentially) join the 3 clauses.

Reply:
Yeah, I wondered about that. Could you do this?

```js
[
 ...data.serviceInfrastructure.containerIds.map((id) => `${CONTAINER_ID}: "${id}"`),
 ...data.serviceInfrastructure.hostNames.map((id) => `${HOSTNAME}: "${id}"`),
 ...data.serviceInfrastructure.podNames.map((id) => `${POD_NAME}: "${id}"`)
].join(' or ')
```

Reply:
Ah, sorry, I'd pushed something else before seeing this. This would be more succinct 🤔  Let me know if it's a deal breaker, I had a nightmare rebuilding Kibana after merging master but I think this should be ready now (bar this change).